src/odemis/driver/xt_client.py

```diff
@@ -2651,6 +2651,7 @@ def __init__(self, name, role, parent, address, **kwargs):
 
     def stop_generate(self):
         logging.debug("Stopping image acquisition")
+        self._genmsg.put(GEN_STOP)
         with self._cancel_access:
             self.cancel_connection._pyroClaimOwnership()
             # Directly calling set_channel_state does not work with XTToolkit.
```

Comment:
If that code, to cancel the current acquisition doesn't work, could you add a big warning in comment that it doesn't work (because reading the code, I had assumed the cancellation worked). Alternatively, you could try to find out how to get working, of course ;-)